qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,591 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Bravyi-Kitaev Super-Fast (BKSF) Mapper."""
+
+from enum import Enum
+from typing import List, Tuple
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators import Pauli, SparsePauliOp
+from qiskit_nature.operators.second_quantization import FermionicOp
+from .fermionic_mapper import FermionicMapper
+
+
+class BravyiKitaevSuperFastMapper(FermionicMapper):
+    """The Bravyi-Kitaev super-fast fermion-to-qubit mapping.
+
+    This implementation follows closely Reference [1].
+
+       References:
+           [1]: Bravyi-Kitaev Superfast simulation of electronic structure on a quantum computer
+                Kanav Setia and James D Whitfield
+                JCP Volume 148, Issue 16 - Published April 2018
+                arXiv:1712.00446
+    """
+
+    def map(self, second_q_op: FermionicOp) -> PauliSumOp:
+        if not isinstance(second_q_op, FermionicOp):
+            raise TypeError("Type ", type(second_q_op), " not supported.")
+
+        edge_list = _bksf_edge_list_fermionic_op(second_q_op)
+        sparse_pauli = _convert_operator(second_q_op, edge_list)
+
+        ## Simplify and sort the result
+        sparse_pauli = sparse_pauli.simplify()
+        indices = sparse_pauli.table.argsort()
+        table = sparse_pauli.table[indices]
+        coeffs = sparse_pauli.coeffs[indices]
+        sorted_sparse_pauli = SparsePauliOp(table, coeffs)
+
+        return PauliSumOp(sorted_sparse_pauli)
+
+
+class TermType(Enum):
+    """Denotes the type of interaction of a Fermionic operator."""
+
+    NUMBER = 1
+    EXCITATION = 2
+    DOUBLE_EXCITATION = 3
+    NUMBER_EXCITATION = 4
+    COULOMB_EXCHANGE = 5
+
+
+def _convert_operator(ferm_op: FermionicOp, edge_list: np.ndarray) -> SparsePauliOp:
+    """Convert a fermionic operator together with qubit-connectivity graph to a Pauli operator.
+
+    This is the heart of the implementation of BKSF mapping. The connectivity graph must be
+    computed before this method is called. The returned Pauli operator must be sorted and simplified.
+
+    Args:
+      `ferm_op`: The fermionic operator to convert.
```

Comment:
Is there are reason to quote the names of the args here ? e.g. 
```
`ferm_op`: ...
```
Normally this would be 
```
ferm_op: ...
```

Reply:
Yes, I think the latter is the usual way to do it.